cpp/daal/src/algorithms/service_kernel_math.h

```diff
@@ -263,6 +270,397 @@ class EuclideanDistances : public PairwiseDistances<FPType, cpu>
     TArray<FPType, cpu> normBufferB;
 };
 
+template <typename FPType, CpuType cpu>
+class MinkowskiDistances : public PairwiseDistances<FPType, cpu>
+{
+public:
+    MinkowskiDistances(const NumericTable & a, const NumericTable & b, const bool powered = true, const double p = 2.0)
+        : _a(a), _b(b), _powered(powered), _p(p)
+    {}
+
+    virtual ~MinkowskiDistances() {}
+
+    virtual services::Status init()
+    {
+        services::Status s;
+        return s;
+    }
+
+    virtual services::Status computeBatch(const FPType * const a, const FPType * const b, size_t aOffset, size_t aSize, size_t bOffset, size_t bSize,
+                                          FPType * const res)
+    {
+        computeBatchImpl(a, b, aOffset, aSize, bOffset, bSize, res);
+
+        return services::Status();
+    }
+
+protected:
+    FPType computeDistance(const FPType * x, const FPType * y, const size_t n)
```

Comment:
Usage of sFasbs, sPowx and loops like this can bring perf results worse even than in Scikit-learn on systems less that AVX-512.

We don't need to solve this at the moment, but need to look after.